static/metrics/agent.go

```diff
@@ -227,9 +230,16 @@ func (a *Agent) ApplyConfig(cfg Config) error {
 	a.mut.Lock()
 	defer a.mut.Unlock()
 
-	if util.CompareYAML(a.cfg, cfg) {
+	newConfigByteArr, err := yaml.Marshal(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to marshal new config: %w", err)
+	}
+	newConfig := string(newConfigByteArr[:])
```

Comment:
Is there a reason that we're doing `string(newConfigByteArr[:])` instead of `string(newConfigByteArr)`? I thought that `[:]` acted as a no-op, but maybe it's doing something I'm not aware of?

Reply:
I copied it from a StackOverflow [discussion](https://stackoverflow.com/questions/40632802/how-to-convert-byte-array-to-string-in-go). Now I realise that the `[:]` is only really useful for arrays. Since here we have a slice, we can indeed remove the `[:]`.